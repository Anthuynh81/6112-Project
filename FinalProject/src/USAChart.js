import React, {memo, useEffect, useState} from "react";
import {
    ZoomableGroup,
    ComposableMap,
    Geographies,
    Geography
} from "react-simple-maps";
import Axios from 'axios';
import { json } from 'd3-fetch';

const geoUrl =   "https://raw.githubusercontent.com/zcreativelabs/react-simple-maps/master/topojson-maps/world-110m.json";
const StartDate = "01/21/2020";

const USAChart = ({ setTooltipContent }) => {
    const [data, setData] = useState([]);


    useEffect(() => {
        Axios.get("http://localhost:5000/USA").then((response) =>{
            setData(response.data.filter(o => o.Updated === StartDate))
        });
    }, []);

    return (
        <>
            <ComposableMap data-tip="" projection="geoAlbersUsa" width={1000} height={800}>
                <ZoomableGroup zoom={1}>>
                    <Geographies geography={geoUrl}>
                        {({ geographies }) =>
                            geographies.filter(geo => geo.name === "United States of America").map(geo => (
                                <Geography
                                    key={geo.rsmKey}
                                    geography={geo}
                                    onMouseEnter={() => {
                                        const NAME = geo.properties.name;
                                        const state= data.find(o => o.AdminRegion1 === NAME);
                                        setTooltipContent(`${geo.properties.name}</br>
                                                        Confirmed - ${state.Confirmed}</br>`);
                                    }}
                                    onMouseLeave={() => {
                                        setTooltipContent("");
                                    }}
                                    style={{
                                        default: {
                                            fill: "#D6D6DA",
                                            outline: "none"
                                        },
                                        hover: {
                                            fill: "#F53",
                                            outline: "none"
                                        },
                                        pressed: {
                                            fill: "#E42",
                                            outline: "none"
                                        }
                                    }}
                                />
                            ))
                        }
                    </Geographies>
                </ZoomableGroup>
            </ComposableMap>
        </>
    );
};

export default memo(USAChart);
