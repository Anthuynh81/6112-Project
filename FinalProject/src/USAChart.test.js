import Axios from 'axios';
import renderer from "react-test-renderer";
import USAChart from "./USAChart";

const testDate = "03/14/2020"


async function getValue() {
    const response = await Axios.get("http://localhost:5000/USA", { params: {date: testDate}})
    return response
}

test("Get request", async () => {
    const response = await getValue()
    //Success code for get request
    expect(response.status).toEqual(200)
})

test("Date filtering", async () => {
    const response = await getValue()
    //Make sure the data of the data being pulled is correct
    expect(response.data[0].Updated).toEqual(testDate)
})

test("Getting correct data", async () => {
    const response = await getValue()
    // Should be empty as it should have no state information
    expect(response.data[0].AdminRegion1).toBe("")
    // Should be empty as it should have no county information
    expect(response.data[0].AdminRegion2).toBe("\r")
})

test('returns null when map is not USA', () => {
    const component = renderer.create(
        <USAChart setTooltipContent={null} map={"County"} date={testDate} toolTip={null}/>
    );
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})

test('returns the chart when map is USA', () => {
    const component = renderer.create(
        <USAChart setTooltipContent={null} map={"USA"} date={testDate} toolTip={null}/>
    );
    let tree = component.toJSON()
    expect(tree.length).toBeGreaterThan(1);
})
