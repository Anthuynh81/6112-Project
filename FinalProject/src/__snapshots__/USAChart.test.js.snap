// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns null when map is not USA 1`] = `null`;
